src/sentry/utils/query.py

```diff
@@ -217,39 +218,17 @@ def get_total_count(self):
         return cursor.fetchone()[0]
 
 
-class WithProgressBar:
-    def __init__(self, iterator, count=None, caption=None):
-        if count is None and hasattr(iterator, "__len__"):
-            count = len(iterator)
+class WithProgressBar[V]:
+    def __init__(
+        self, iterator: Iterable[V], count: int | None = None, caption: str | None = None
+    ) -> None:
         self.iterator = iterator
         self.count = count
-        self.caption = str(caption or "Progress")
+        self.caption = caption or "Progress"
 
-    def __iter__(self):
-        pbar = progressbar.ProgressBar(
-            widgets=[
-                f"{self.caption}: ",
-                progressbar.Percentage(),
-                " ",
-                progressbar.Bar(),
-                " ",
-                progressbar.ETA(),
-            ],
-            max_value=self.count,
-            # The default update interval is every 0.1s,
-            # which for large migrations would easily logspam GoCD.
-            min_poll_interval=10,
```

Comment:
of note: click's progress bar doesn't need this because it will only produce an update if the bar changes